modules/core/include/opencv2/core/cvstd_wrapper.hpp

```diff
@@ -105,27 +105,30 @@ struct Ptr : public std::shared_ptr<T>
     template<class Y, class Deleter>
     void reset(Y* ptr, Deleter d) { std::shared_ptr<T>::reset(ptr, d); }
 
-    void reset() noexcept { std::shared_ptr<T>::reset(); }
+    void reset() CV_NOEXCEPT { std::shared_ptr<T>::reset(); }
 
     Ptr& operator=(const Ptr& o) { std::shared_ptr<T>::operator =(o); return *this; }
     template<typename Y> inline Ptr& operator=(const Ptr<Y>& o) { std::shared_ptr<T>::operator =(o); return *this; }
 
-    T* operator->() const noexcept { return std::shared_ptr<T>::get();}
-    typename std::add_lvalue_reference<T>::type operator*() const noexcept { return *std::shared_ptr<T>::get(); }
+    T* operator->() const CV_NOEXCEPT { return std::shared_ptr<T>::get();}
+    typename std::add_lvalue_reference<T>::type operator*() const CV_NOEXCEPT { return *std::shared_ptr<T>::get(); }
 
     // OpenCV 3.x methods (not a part of standart C++ library)
     inline void release() { std::shared_ptr<T>::reset(); }
     inline operator T* () const { return std::shared_ptr<T>::get(); }
-    inline bool empty() const { return std::shared_ptr<T>::get() == NULL; }
+    inline bool empty() const { return std::shared_ptr<T>::get() == nullptr; }
+
+    inline bool operator != (const size_t &addr) { return std::shared_ptr<T>::get() != (T*)addr; }
+    inline bool operator == (const size_t &addr) { return std::shared_ptr<T>::get() == (T*)addr; }
```

Comment:
Lets avoid these dangerous operators.
Such compare operators are out of scope of standard `std::shared_ptr` (only `nullptr_t` is supported, but it is not used in OpenCV code yet - I saw raw zeros `0` or `NULL` only).